src/caja-pathbar.c

```diff
@@ -1541,6 +1541,10 @@ setup_file_path_mounted_mount (GFile *location, ButtonData *button_data)
     result = FALSE;
     volume_monitor = g_volume_monitor_get ();
     mounts = g_volume_monitor_get_mounts (volume_monitor);
+    /* Return false if button has not been set up yet or has been destroyed*/
```

Comment:
I would place this closer to the beginning, because `mounts` apparently should be freed (see `g_list_free_full` at the end of the function), so it's better to return before it's allocated.

Reply:
This might also go into a separate commit, as it's not related to HiDPI, and I'd pick it for 1.18 branch then.

Reply:
Hmm, maybe it won't help in 1.18... `button_data->button` is only used to get the scale.

Reply:
This is actually needed here, as one of the warnings was coming from it. Otherwise it would be a separate PR. Moving though as you suggested